FSharp-4.5/FS-1053-span.md

```diff
@@ -336,6 +336,8 @@ in some situations where capture must be avoided.
 * Cannot be static or instance members of a class or normal struct
 * Cannot be captured by any closure construct (async methods or lambda expressions)
 * Cannot be used as a generic parameter
+    - In F# 9 and higher, this restriction is relaxed if the generic parameter is defined in C# using the `allows ref struct` anti-constraint. F# can instantiate such generics in types and methods with byref-like types. As a few examples, this affects BCL delegate types (`Action<>`,`Func<>`), interfaces (`IEnumerable<>`,`IComparable<>`) and generic arguments with a user-provided accumulator function (`String.string Create<TState>(int length, TState state, SpanAction<char, TState> action)`)
```

Comment:
I can make it pretend to be an attribute perharps?
I have the same concern - making it "look like" F# syntax would be just confusing, since such syntax would not be valid.

Reply:
I wonder if we can/should just make it a "note"

Reply:
To be fair, the constraint information shown in the tooling already sometimes deviates from actual F# syntax:

![image](https://github.com/user-attachments/assets/5682cb3e-e7fb-4f36-bc38-8fbee8080502)

Reply:
Agreed @brianrourkeboll . I went with displaying it using the C# syntax, i.e. "allows ref struct".